skimage/measure/_find_contours.py

```diff
@@ -111,12 +117,17 @@ def find_contours(array, level,
     if array.ndim != 2:
         raise ValueError('Only 2D arrays are supported.')
     level = float(level)
+    if mask is not None:
+        mask = np.asarray(mask, dtype=np.uint8)
+        if mask.shape != array.shape:
+            raise ValueError('Parameters "array" and "mask" must have same shape.')
```

Comment:
Could you add a test that tests this case?

Reply:
Done! See 362dfb982